### PR TITLE
[Admin][ProductReview] Introduce reviews in admin panel

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | 
| License         | MIT

Features:
- [x] Browsing product reviews
- [x] Editing product reviews
- [x] Accepting and rejecting product reviews on the list
- [x] Deleting product reviews
- [x] Deleting a product with its reviews
- [x] Validation for editing a product review

### CODE REVIEW

File: features/product/managing_product_reviews/editing_product_review.feature [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    Background:
+        Given the store has customer "Mike Ross" with email "ross@teammike.com"
+        And the store has a product "Lamborghini Gallardo Model"
+        And this product has a review titled "Awesome" and rated 4 with a comment "Nice product" added by customer "ross@teammike.com"
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Changing a title of a product review
+        When I want to modify the "Awesome" product review
+        And I change its title to "Bewildering"
```
Comment (michalmarcinkowski) [inline features/product/managing_product_reviews/editing_product_review.feature]: Maybe we should give an example with correcting a typo in the title?

File: features/product/managing_product_reviews/editing_product_review.feature [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        Then I should be notified that it has been successfully edited
+        And this product review comment should be "Really nice product!"
+
+    @ui @javascript
+    Scenario: Changing a rating of a product review
+        When I want to modify the "Awesome" product review
+        And I choose 5 as its rating
+        And I save my changes
+        Then I should be notified that it has been successfully edited
+        And this product review rating should be 5
```
Comment (michalmarcinkowski) [inline features/product/managing_product_reviews/editing_product_review.feature]: Do we have a scenario for changing the rating and verifying that the average rating is recalculated?

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/product_review.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+sylius_admin_product_review:
+    resource: |
+        alias: sylius.product_review
+        section: admin
+        templates: SyliusAdminBundle:Crud
+        except: ['show', 'create']
+        redirect: update
+        grid: sylius_admin_product_review
+        permission: true
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/product_review.yml]: Redundant here are everywhere else.

File: src/Sylius/Bundle/AdminBundle/Resources/views/ProductReview/Grid/Action/reject.html.twig [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{% if 'new' == data.status %}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AdminBundle/Resources/views/ProductReview/Grid/Action/reject.html.twig]: Same as above.

File: features/product/managing_product_reviews/editing_product_review.feature (hunk 2 above)
Comment (GSadee) [inline features/product/managing_product_reviews/editing_product_review.feature]: But for now, we don't display the average rating in admin. Should I add this somewhere to could check that?

File: features/product/managing_product_reviews/editing_product_review.feature (hunk 1 above)
Comment (pamil) [inline features/product/managing_product_reviews/editing_product_review.feature]: It makes this scenarios making harder to understand at the first sight, at least for me.

File: src/Sylius/Behat/Context/Setup/ProductReviewContext.php [hunk 5]
```diff
@@ -75,16 +75,17 @@ public function productHasAReview(ProductInterface $product)
     }
 
     /**
-     * @Given /^(this product) has(?:| also) a review titled "([^"]+)" and rated (\d+) added by (customer "[^"]+")(?:|, created (\d+) days ago)$/
+     * @Given /^(this product) has(?:| also) a review titled "([^"]+)" and rated (\d+)(?:| with a comment "([^"]+)") added by (customer "[^"]+")(?:|, created (\d+) days ago)$/
```
Comment (pamil) [inline src/Sylius/Behat/Context/Setup/ProductReviewContext.php]: `This product has a review title "Foo" and rated 3 added by customer "Bar"` won't work, too many optional placeholders.

Comment (pamil) [inline src/Sylius/Behat/Context/Setup/ProductReviewContext.php]: Anyway, this step is way too long, comment / created at settings may be made independent steps.

File: src/Sylius/Behat/Context/Setup/ProductReviewContext.php [hunk 6]
```diff
@@ -93,15 +94,16 @@ public function thisProductHasAReviewTitledAndRatedAddedByCustomer(
     }
 
     /**
-     * @Given /^(this product) has(?:| also) a review titled "([^"]+)" and rated (\d+) added by (customer "[^"]+") which is not accepted yet$/
+     * @Given /^(this product) has(?:| also) a review titled "([^"]+)" and rated (\d+)(?:| with a comment "([^"]+)") added by (customer "[^"]+") which is not accepted yet$/
```
Comment (pamil) [inline src/Sylius/Behat/Context/Setup/ProductReviewContext.php]: Same as for another factory step above.

**Comment by pjedrzejewski** [comment]
Thank you Grzesiu, nice work! :)

**Pull request merged**